File: app/src/main/java/com/github/windsekirun/yukarisynthesizer/di/AppBindsModule.java
package com.github.windsekirun.yukarisynthesizer.di;

import android.app.Application;
import com.github.windsekirun.yukarisynthesizer.MainApplication;
import dagger.Binds;
import dagger.Module;

@Module
public abstract class AppBindsModule {

    @Binds
    abstract Application bindApplication(MainApplication application);
}

File: app/src/main/java/com/github/windsekirun/yukarisynthesizer/di/AppBindsModule.kt
package com.github.windsekirun.yukarisynthesizer.di

import android.app.Application
import com.github.windsekirun.yukarisynthesizer.MainApplication
import dagger.Binds
import dagger.Module

/**
 * Dagger module for provide project-specific class
 */
@Module
abstract class AppBindsModule {

    @Binds
    abstract fun bindApplication(application: MainApplication): Application
}